### PR TITLE
Make update commands upserty

### PR DESCRIPTION
**Pull request by jdoliner**

### CODE REVIEW

File: src/server/pps/server/api_server.go [hunk 1]
```diff
@@ -2806,6 +2808,18 @@ func (a *apiServer) updatePipelineSpecCommit(pachClient *client.APIClient, pipel
 	return err
 }
 
+func createSpecRepo(c *client.APIClient) error {
+	_, err := c.PfsAPIClient.CreateRepo(
+		c.Ctx(),
+		&pfs.CreateRepoRequest{
+			Repo:        client.NewRepo(ppsconsts.SpecRepo),
+			Update:      true,
```
Comment (msteffen) [inline src/server/pps/server/api_server.go]: If I'm reading this PR correctly, one of the consequences of this changes is that the `CreateRepo(__spec__)` call that all pachd pods (including e.g. sidecars) make on startup will now actually update the spec repo.

I'm not sure if there's any problem with that with that (more writes to etcd, I guess?), but it seems notable enough to confirm?

Comment (jdoliner) [reply src/server/pps/server/api_server.go]: That is true and something occurred to me while implementing this. In general I'm not crazy about it, although before this existed we were already sending a read request to etcd, but now it's a read and a write. Maybe we should make it so update doesn't actually do the write if nothing is changing, as will be the case with these updates.

Comment (msteffen) [reply src/server/pps/server/api_server.go]: Hmm, I mean if it's not a ton of work that definitely seems like a useful optimization

Comment (jdoliner) [reply src/server/pps/server/api_server.go]: Sounds good, shouldn't be much work.

**Pull request merged**